python/pyspark/sql/utils.py

```diff
@@ -77,6 +83,8 @@ def deco(*a, **kw):
                 raise QueryExecutionException(s.split(': ', 1)[1], stackTrace)
             if s.startswith('java.lang.IllegalArgumentException: '):
                 raise IllegalArgumentException(s.split(': ', 1)[1], stackTrace)
+            if s.startswith('java.lang.NoClassDefFoundError: '):
```

Comment:
I think it is still helpful. This is to hide the long java stack trace for improving the error reporting. It was introduced in the PR https://github.com/apache/spark/pull/7135.